components/canvas/NodeTooltip.tsx

```diff
@@ -9,24 +15,56 @@ type NodeTooltipContainerProps = {
   isVisible?: boolean;
   position?: Position;
   style?: NodeToolbarProps["style"];
+  isEditing?: boolean;
+  nodeRef?: RefObject<HTMLDivElement>;
 };
 
 export const NodeTooltipContainer = ({
   children,
   isVisible,
   position,
   style,
+  isEditing,
+  nodeRef,
 }: NodeTooltipContainerProps) => {
+  const toolTipRef = useRef<HTMLDivElement>(null);
+  const [tooltipPosition, settooltipPosition] = useState<Position | undefined>(
+    undefined
+  );
+  const [offset, setoffset] = useState(10);
+  useEffect(() => {
```

Comment:
We get some stuttering since reactflow defaults to top position, but we quickly change it to bottom position after the calculation is done. We can change to useLayoutEffect to make sure the positioning logic runs before the browser repaint.